addons/account/tests/test_fiscal_position.py

```diff
@@ -0,0 +1,31 @@
+from openerp.tests.common import TransactionCase
+
+class TestFiscalPosition(TransactionCase):
+    """Tests for fiscal positions (account.fiscal.position).
+    """
+
+    def setUp(self):
+        super(TestFiscalPosition, self).setUp()
+        self.fiscal_position_model = self.registry('account.fiscal.position')
+        self.res_partner_model = self.registry('res.partner')
+        self.res_country_model = self.registry('res.country')
+
+    def test_fiscal_position(self):
+        cr, uid = self.cr, self.uid
+        company_id = 1
+        country_id = self.res_country_model.search(cr, uid, [('name', '=', 'France')])[0]
+        partner_id = self.res_partner_model.create(cr, uid, dict(
+                                                   name="George",
+                                                   vat_subjected=True,
+                                                   notify_email="always",
+                                                   country_id=country_id))
+        fp1_id = self.fiscal_position_model.create(cr, uid, dict(name="EU-VAT-FR",
+                                                                 auto_apply=True,
+                                                                 country_id=country_id,
+                                                                 vat_required=False))
+        fp2_id = self.fiscal_position_model.create(cr, uid, dict(name="EU-VAT-FR",
```

Comment:
Also you could force the sequences of the 2 positions, forcing B2C as the highest priority, ensuring the order of the match is deterministic, and not just due to luck ;-)